src/Features/Core/Portable/EditAndContinue/EditSession.cs

```diff
@@ -479,88 +576,260 @@ private async Task<ProjectChanges> GetProjectChangesAsync(Project project, Cance
             {
                 throw ExceptionUtilities.Unreachable;
             }
         }
 
+        private static unsafe bool CreateInitialBaselineForDeferredModuleUpdate(
+            Project project,
+            ArrayBuilder<(ProjectId, ImmutableArray<Diagnostic>)> diagnostics,
+            out EmitBaseline baseline,
+            out PEReader peReader,
+            out MetadataReaderProvider pdbReaderProvider)
+        {
+            // Since the module has not been loaded to the debuggee the debugger does not have its metadata or symbols available yet.
+            // Read the metadata and symbols from the disk. Close the files as soon as we are done emitting the delta to minimize 
+            // the time when they are being locked. Since we need to use the baseline that is produced by delta emit for the subsequent
+            // delta emit we need to keep the module metadata and symbol info backing the symbols of the baseline alive in memory. 
+            // Alternatively, we could drop the data once we are done with emitting the delta and re-emit the baseline again 
+            // when we need it next time and the module is loaded.
+
+            baseline = null;
+            peReader = null;
+            pdbReaderProvider = null;
+
+            // TODO: use obj path
+            bool success = false;
+            string modulePath = project.OutputFilePath;
+            string fileBeingRead = modulePath;
+            using (var peFileStream = FileUtilities.OpenFileStream(modulePath))
+            {
+                try
+                {
+                    // Let PEReader hold on the memory backing the metadata blob even after the file stream is closed.
+                    peReader = new PEReader(peFileStream, PEStreamOptions.PrefetchMetadata | PEStreamOptions.LeaveOpen);
+                    var metadataBlock = peReader.GetMetadata();
+
+                    var moduleMetadata = ModuleMetadata.CreateFromMetadata((IntPtr)metadataBlock.Pointer, metadataBlock.Length);
+                    if (peReader.TryOpenAssociatedPortablePdb(modulePath, f => FileUtilities.OpenFileStream(fileBeingRead = f), out pdbReaderProvider, out var pdbPath))
+                    {
+                        // Portable or Embedded PDB
+                        var pdbReader = pdbReaderProvider.GetMetadataReader();
+                        var infoReader = EditAndContinueMethodDebugInfoReader.CreatePortable(pdbReader);
+
+                        baseline = EmitBaseline.CreateInitialBaseline(
+                            moduleMetadata,
+                            infoReader.GetDebugInfo,
+                            infoReader.GetLocalSignature,
+                            infoReader.IsPortable);
+                    }
+                    else
+                    {
+                        // Windows PDB
+                        throw new NotImplementedException("TODO");
```

Comment:
❔ Bug for this? Is this a new limitation or one that existed in the old implementation as well?